TELCODOCS-1957: Hardware accelerator overview (#82500)
openshift/openshift-docs: hardware_accelerators/about-hardware-accelerators.adoc

:_mod-docs-content-type: ASSEMBLY
[id="about-hardware-accelerators"]
= About hardware accelerators
include::_attributes/common-attributes.adoc[]
:context: gpu-accelerators

toc::[]

[IMPORTANT]
====
Hardware accelerators are a Technology Preview feature only. Technology Preview features are not supported with Red{nbsp}Hat production service level agreements (SLAs) and might not be functionally complete. Red{nbsp}Hat does not recommend using them in production. These features provide early access to upcoming product features, enabling customers to test functionality and provide feedback during the development process.

For more information about the support scope of Red{nbsp}Hat Technology Preview features, see link:https://access.redhat.com/support/offerings/techpreview/[Technology Preview Features Support Scope].
====

> 

Reviewer: Is there a reason we are calling this tech preview?  All of our accelerator tools like KMM, NFD and DTK are fully supported.  @egallen thoughts?

> * Tensor processing units (TPUs) 

Reviewer 2: Could we remove this TPU line?

// GRAPHIC HERE

Red{nbsp}Hat OpenShift AI is an open hybrid AI/ML platform built on Red{nbsp}Hat OpenShift that enables enterprises to create and deliver AI-enabled applications securely and at scale across hybrid clouds. It enables data acquisition and preparation, model training and fine-tuning, model serving and model monitoring, and hardware acceleration. 

// Link to RHOAI landing page

Specialized hardware accelerators are designed to work within air gapped (disconnected) environments. This is because of customer requirements to maintain a secure environment for development and testing.  

Hardware Operators use the operating framework of a Kubernetes cluster to enable the required accelerator resources. A device plugin is also provided that can be deployed manually or as a DaemonSet. This plugin enables the registration of the GPU in the cluster.   

== Supported vendors

Red{nbsp}Hat Openshift supports hardware accelerators provided by the following vendors:

* AMD
* NVIDIA
* Intel

> 

Reviewer: Do we support these for OpenShift?  I don't think so.  We provide accelerator tools.  The partners are responsible for their operators which integrate with our accelerator tools.  

RHEL AI is a different topic.  In this case Red Hat does support the partner accelerator specific images and content.

@egallen thoughts?

Reviewer 2: Yes, better to remove the word support.

:_mod-docs-content-type: ASSEMBLY
[id="about-hardware-accelerators"]
= About hardware accelerators
include::_attributes/common-attributes.adoc[]
:context: about-hardware-accelerators

toc::[]

> Specialized hardware accelerators play a key role in the emerging generative artificial intelligence and machine learning (AI/ML) industry. Specifically, hardware accelerators are essential to the training and serving of large language and other foundational models that power this new technology. It allows data scientists, data engineers, ML engineers, and developers to take advantage of the specialized hardware acceleration for data-intensive transformations and model development and serving. Much of that ecosystem is open source with a number of contributing partners and open source foundations.

Reviewer 3: ```suggestion
Specialized hardware accelerators play a key role in the emerging generative artificial intelligence and machine learning (AI/ML) industry. Specifically, hardware accelerators are essential to the training and serving of large language and other foundational models that power this new technology. Data scientists, data engineers, ML engineers, and developers can take advantage of the specialized hardware acceleration for data-intensive transformations and model development and serving. Much of that ecosystem is open source, with a number of contributing partners and open source foundations.
```
Is (AI/ML) an approved abbreviation? I see AI listed as approved in [IBM Style](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#AI), but I don't see ML mentioned in either IBM Style or the Supplementary Style Guide, and I don't see the combination of "AI/ML" approved.

> Red{nbsp}Hat OpenShift provides support for cards and peripheral hardware that add processing units that comprise hardware accelerators:

Reviewer 3: ```suggestion
Red{nbsp}Hat OpenShift supports cards and peripheral hardware that add processing units that comprise hardware accelerators:
```

* Graphical processing units (GPUs)
* Neural processing units (NPUs)
* Application-specific integrated circuits (ASICs)
* Data processing units (DPUs)
* Others

image::OCP_HW_Accelerators_4.png[Supported hardware accelerators cards and peripherals]

> Specialized hardware accelerators provide a rich set of benefits for AI/ML development:

Reviewer 3: See my earlier comment about "AI/ML".

> One platform for all:: Allows for collaborative environments for developers, data engineers, data scientists, and DevOps.

Reviewer 3: ```suggestion
One platform for all:: A collaborative environment for developers, data engineers, data scientists, and DevOps
```
When possible, avoid anthropomorphic language: https://www.ibm.com/docs/en/ibm-style?topic=grammar-anthropomorphism

> Extended capabilities with Operators:: Operators allow for bringing AI/ML capabilities to OpenShift.

Reviewer 3: ```suggestion
Extended capabilities with Operators:: Operators provide AI and ML capabilities to {product-title}
```

> Hybrid-cloud support:: On-premise support for model development, delivery, and deployment. 

Reviewer 3: ```suggestion
Hybrid-cloud support:: On-premise support for model development, delivery, and deployment
```

> Support for AI/ML workloads:: Model testing, iteration, integration, promotion, and serving into production as services.

Reviewer 3: ```suggestion
Support for AI and ML workloads:: Model testing, iteration, integration, promotion, and serving into production as services
```
This is a little hard to follow. Does the phrase "as services" modify only "serving into production", or does it modify "model testing, iteration, integration, promotion, and serving into production"?

> Red{nbsp}Hat provides an optimized platform to enable these specialized hardware accelerators in our RHEL and OpenShift platforms at the Linux (kernel and userspace) and Kubernetes layers. To do this, Red{nbsp}Hat combines the proven capabilities of Red{nbsp}Hat OpenShift AI and Red{nbsp}Hat OpenShift in a single enterprise-ready AI application platform. 

Reviewer 3: ```suggestion
Red{nbsp}Hat provides an optimized platform to enable these specialized hardware accelerators in {op-system-base-full} and {product-title} platforms at the Linux (kernel and userspace) and Kubernetes layers. To do this, Red{nbsp}Hat combines the proven capabilities of Red{nbsp}Hat OpenShift AI and {product-title} in a single enterprise-ready AI application platform. 
```

> Hardware Operators use the operating framework of a Kubernetes cluster to enable the required accelerator resources. A device plugin is also provided that can be deployed manually or as a DaemonSet. This plugin enables the registration of the GPU in the cluster. 

Reviewer 4: Suggesting for better readability / getting rid of passive voice. Also a suggestion in the second sentence, but not sure if that changes the meaning

```suggestion
Hardware Operators use the operating framework of a Kubernetes cluster to enable the required accelerator resources. You can also deploy the provided device plugin manually or as a daemon set. This plugin registers the GPU in the cluster. 
```

> Certain specialized hardware accelerators are designed to work within air gapped (disconnected) environments where a secure environment must be maintained for development and testing.  

Reviewer 4: We prefer "disconnected" in OCP, so just that should be fine

https://redhat-documentation.github.io/supplementary-style-guide/#disconnected

```suggestion
Certain specialized hardware accelerators are designed to work within disconnected environments where a secure environment must be maintained for development and testing.  
```

> Certain specialized hardware accelerators are designed to work within air gapped (disconnected) environments. This is because of customer requirements to maintain a secure environment for development and testing.  

Reviewer 3: ```suggestion
Certain specialized hardware accelerators are designed to work within air gapped (disconnected) environments, where a secure environment must be maintained for development and testing.  
```
Trying to avoid starting a sentence with an ambiguous pronoun (This). See https://www.ibm.com/docs/en/ibm-style?topic=grammar-pronouns#ambiguous-pronoun-references

[id="hardware-accelerators"]
== Hardware accelerators

> Red{nbsp}Hat Openshift enables the following hardware accelerators:

Reviewer 4: Same about using {product-title}

* NVIDIA GPU
* AMD Instinct GPU

> * Intel Gaudi

Reviewer 3: ```suggestion
* Intel(R) Gaudi(R)
```
Mark registered trademark symbols on their first instance.

[role="_additional-resources"]
.Additional resources

* link:https://docs.redhat.com/en/documentation/red_hat_openshift_ai_self-managed/2-latest/html/introduction_to_red_hat_openshift_ai/index[Introduction to Red Hat OpenShift AI]

* link:https://docs.nvidia.com/datacenter/cloud-native/openshift/latest/index.html[
NVIDIA GPU Operator on Red Hat OpenShift Container Platform]

* link:https://www.amd.com/en/products/accelerators/instinct.html[AMD Instinct Accelerators]

> * link:https://www.intel.com/content/www/us/en/products/details/processors/ai-accelerators/gaudi-overview.html[Intel® Gaudi® Al Accelerators]

Reviewer 3: ```suggestion
* link:https://www.intel.com/content/www/us/en/products/details/processors/ai-accelerators/gaudi-overview.html[Intel Gaudi Al Accelerators]
```

* Extended capabilities with Operators. Operators allow for bringing AI/ML capabilities to OpenShift.
* Hybrid-cloud support. On-premise support for model development, delivery, and deployment. 
* Support for AI/ML workloads. Model testing, iteration, integration, promotion, and serving into production as services.

Red{nbsp}Hat provides an optimized platform to enable these specialized hardware accelerators in our RHEL and OpenShift platforms at the Linux (kernel and userspace) and Kubernetes layers. To do this, Red{nbsp}Hat combines the proven capabilities of Red{nbsp}Hat OpenShift AI and Red{nbsp}Hat OpenShift in a single enterprise-ready AI application platform. 

image::OCP_HW_Accelerators_2.png[OCP hardware accelerators]

Red{nbsp}Hat OpenShift AI is an open hybrid AI/ML platform built on Red{nbsp}Hat OpenShift that enables enterprises to create and deliver AI-enabled applications securely and at scale across hybrid clouds. It enables data acquisition and preparation, model training and fine-tuning, model serving and model monitoring, and hardware acceleration. 

> Specialized hardware accelerators are designed to work within air gapped (disconnected) environments. This is because of customer requirements to maintain a secure environment for development and testing.  

Reviewer 2: Could you change the sentence to "Certain specialized hardware accelerators are designed to work within air gapped (disconnected) environments."

Certain specialized hardware accelerators are designed to work within air gapped (disconnected) environments. This is because of customer requirements to maintain a secure environment for development and testing.  

== Supported AI accelerators

> Red{nbsp}Hat Openshift supports the following AI accelerators:

Reviewer 2: we can replace "supports" by "enables"

:_mod-docs-content-type: ASSEMBLY
[id="about-hardware-accelerators"]
= About hardware accelerators
include::_attributes/common-attributes.adoc[]

> :context: gpu-accelerators

Reviewer 5: This should be the same as the anchor ID: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#assembly-file-metadata

toc::[]

> Specialized hardware accelerators play a key role in the emerging generative artificial intelligence and machine learning (AI/ML) industry. Specifically, hardware accelerators are essential to the training and serving of large language and other foundational models that power this new technology. It allows data scientists, data engineers, ML engineers, and developers to take advantage of the specialized hardware acceleration for data-intensive transformations and model development and serving. Much of that ecosystem is open source with a number of contributing partners and open source foundations.

Reviewer 5: This assembly should be [modularized](https://redhat-documentation.github.io/modular-docs/). At the moment, it looks like a concept module followed by a reference module to me.

> Red{nbsp}Hat OpenShift provides support for cards and peripheral hardware that add processing units that comprise hardware accelerators:

Reviewer 5: Remember to use the product title attribute rather than "OpenShift" in most cases.

Reviewer 3: +1 to this comment from Max. Also, do you mean Red Hat OpenShift, or Red Hat OpenShift Container Platform?

* Graphical processing units (GPUs)
* Neural processing units (NPUs)
* Application-specific integrated circuits (ASICs)
* Data processing units (DPUs)
* Others

> image::OCP_HW_Accelerators_4.png[OCP hardware accelerators]

Reviewer 5: Alt text needs some fleshing out to be equivalent to the image.

Extended capabilities with Operators:: Operators allow for bringing AI/ML capabilities to OpenShift.
Hybrid-cloud support:: On-premise support for model development, delivery, and deployment. 
Support for AI/ML workloads:: Model testing, iteration, integration, promotion, and serving into production as services.

Red{nbsp}Hat provides an optimized platform to enable these specialized hardware accelerators in our RHEL and OpenShift platforms at the Linux (kernel and userspace) and Kubernetes layers. To do this, Red{nbsp}Hat combines the proven capabilities of Red{nbsp}Hat OpenShift AI and Red{nbsp}Hat OpenShift in a single enterprise-ready AI application platform. 

Hardware Operators use the operating framework of a Kubernetes cluster to enable the required accelerator resources. A device plugin is also provided that can be deployed manually or as a DaemonSet. This plugin enables the registration of the GPU in the cluster. 

Certain specialized hardware accelerators are designed to work within air gapped (disconnected) environments. This is because of customer requirements to maintain a secure environment for development and testing.  

> == Hardware accelerators

Reviewer 5: Missing anchor ID.

Red{nbsp}Hat Openshift enables the following hardware accelerators:

* NVIDIA GPU
* AMD Instinct GPU
* Intel Gaudi

> .Prerequisites

Reviewer 5: Prerequisites in this style are used for procedure modules. Is there a task that should follow here?
